lms/djangoapps/discussion_api/api.py

```diff
@@ -43,15 +43,15 @@
 from openedx.core.djangoapps.course_groups.cohorts import get_cohort_id
 
 
-def _get_course_or_404(course_key, user):
+def _get_course_or_raise_not_found(course_key, user):
     """
-    Get the course descriptor, raising Http404 if the course is not found,
+    Get the course descriptor, raising ObjectDoesNotExist if the course is not found,
     the user cannot access forums for the course, or the discussion tab is
     disabled for the course.
     """
     course = get_course_with_access(user, 'load', course_key, check_if_enrolled=True)
```

Comment:
@wajeeha-khalid I just looked into this a bit more. It turns out `get_course_with_access` raises a 404 (https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/courses.py#L83). So we will have to catch that and raise `CourseNotFoundError` explicitly.